code/modules/events/event_dynamic.dm

```diff
@@ -56,12 +56,19 @@ var/list/event_last_fired = list()
 	if(!spacevines_spawned)
 		possibleEvents[/datum/event/spacevine] = 15
 
-	if(minutes_passed >= 30 && active_with_role["Engineer"] > 1) // Give engineers time to not set up the engine
+	if(active_with_role["Engineer"] > 1)
 		possibleEvents[/datum/event/meteor_wave] = 15
 		possibleEvents[/datum/event/meteor_shower] = 40
-		possibleEvents[/datum/event/thing_storm/blob_shower] = 25
-		possibleEvents[/datum/event/thing_storm/blob_storm] = 10
 		possibleEvents[/datum/event/immovable_rod] = 15
+		possibleEvents[/datum/event/thing_storm/blob_shower] = 25//Blob Cluster
+
+	if((active_with_role["Engineer"] > 1) && (active_with_role["Security"] > 1))
```

Comment:
Did you know that the meteor and immovable rod events are already dependent on there being engineers on the station? Likewise, alien and spider infestations only happen if there are living sec officers, and viral outbreak only happen if there's medical staff.